Packages/com.unity.inputsystem/InputSystem/Utilities/ArrayHelpers.cs

```diff
@@ -176,6 +176,20 @@ public static int IndexOf<TValue>(this TValue[] array, Predicate<TValue> predica
             return -1;
         }
 
+        public static int IndexOf<TValue>(this TValue[] array, Predicate<TValue> predicate, int startIndex = 0, int count = -1)
+        {
+            if (array != null)
+            {
+                var end = startIndex + (count < 0 ? array.Length - startIndex : count);
+                for (var i = startIndex; i < end; ++i)
```

Comment:
Could do with some bounds checks on these indexes

Reply:
Yup, would you suggest I fix all of them (previously existing) since this is lacking for all pre-existing helpers?

Reply:
Probably should be done for all of them that directly index the array, but I'd be ok if that was done in a separate PR so it doesn't muddy this one. Assert will only result in exceptions during development though right, which is where you want them. I don't really understand what you mean by checking twice. I'd add debug asserts for startIndex and end being in range is all.